external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tech.mlsql.plugins.ets
+
+import org.apache.spark.ml.param.Param
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.sql.execution.MLSQLAuthParser
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import streaming.dsl.ScriptSQLExec
+import streaming.dsl.auth.TableAuthResult
+import streaming.dsl.mmlib._
+import streaming.dsl.mmlib.algs.param.WowParams
+import streaming.log.WowLog
+import tech.mlsql.common.utils.log.Logging
+import tech.mlsql.dsl.auth.ETAuth
+import tech.mlsql.dsl.auth.dsl.mmlib.ETMethod.ETMethod
+
+/**
+  * 09/08/2021 hellozepp(lisheng.zhanglin@163.com)
+  */
+class SyntaxAnalyzeExt(override val uid: String) extends SQLAlg with WowParams with Logging with WowLog with ETAuth {
+  def this() = this(Identifiable.randomUID("tech.mlsql.plugins.ets.SyntaxAnalyzeExt"))
+
+  override def train(df: DataFrame, path: String, params: Map[String, String]): DataFrame = {
+    val context = ScriptSQLExec.contextGetOrForTest()
+    val _sql = params(sql.name)
+    params("action") match {
+      case "extractTables" =>
+        import df.sparkSession.implicits._
+        MLSQLAuthParser.filterTables(_sql, context.execListener.sparkSession).map(_.table).toDF("tableName")
+      case _ => throw new NoSuchElementException("Failed to execute SyntaxAnalyzeExt, unsupported statement")
```

Comment:
unsupported statement?  maybe should `unsupported action`? Also ,since you have only one action for now, maybe we can set action `extractTables` as default.

Reply:
Yes it will be better